File: app/src/main/java/com/zibete/proyecto1/utils/SnackUtils.java
package com.zibete.proyecto1.utils;

import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.ColorRes;
import androidx.annotation.DrawableRes;
import androidx.annotation.Nullable;

import com.google.android.material.snackbar.Snackbar;
import com.zibete.proyecto1.R;

public class SnackUtils {

    public static void show(
            View root,
            String message,
            int duration,
            @ColorRes int bgColor,
            @Nullable String actionText,
            @Nullable View.OnClickListener action,
            @DrawableRes int iconRes
    ) {
        Snackbar snackbar = Snackbar.make(root, "", duration);

        // Inflamos nuestro layout
        Snackbar.SnackbarLayout layout = (Snackbar.SnackbarLayout) snackbar.getView();
        layout.setBackgroundColor(0x00000000); // transparente para que se vea el card

        View customView = LayoutInflater.from(root.getContext())
                .inflate(R.layout.layout_snackbar_zibe, null, false);

        // Texto
        TextView tv = customView.findViewById(R.id.snack_text);
        tv.setText(message);

        // Icono opcional
        ImageView iv = customView.findViewById(R.id.snack_icon);
        if (iconRes != 0) {
            iv.setVisibility(View.VISIBLE);
            iv.setImageResource(iconRes);
        } else {
            iv.setVisibility(View.GONE);
        }

        // Acción opcional
        TextView tvAction = customView.findViewById(R.id.snack_action);
        if (actionText != null && action != null) {
            tvAction.setVisibility(View.VISIBLE);
            tvAction.setText(actionText);
            tvAction.setOnClickListener(v -> {
                action.onClick(v);
                snackbar.dismiss();
            });
        }

        // Color de fondo
        customView.setBackgroundTintList(
                root.getResources().getColorStateList(bgColor)
        );

        // Limpiamos views anteriores del layout del snackbar
        layout.removeAllViews();
        layout.addView(customView);

        snackbar.show();
    }

    // Overload rápido para “mensaje + OK”
    public static void showInfo(View root, String message) {
        show(
                root,
                message,
                Snackbar.LENGTH_SHORT,
                R.color.colorC,      // tu color
                "OK",
                v -> {},
                R.drawable.ic_info_24
        );
    }
}

File: app/src/main/java/com/zibete/proyecto1/utils/SnackUtils.kt
package com.zibete.proyecto1.utils

import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.annotation.ColorRes
import androidx.annotation.DrawableRes
import androidx.core.content.ContextCompat
import com.google.android.material.snackbar.Snackbar
import com.google.android.material.snackbar.Snackbar.SnackbarLayout
import com.zibete.proyecto1.R

object SnackUtils {

    @JvmStatic
    fun show(
        root: View,
        message: String,
        duration: Int = Snackbar.LENGTH_SHORT,
        @ColorRes bgColor: Int = R.color.colorC,
        actionText: String? = null,
        action: ((View) -> Unit)? = null,
        @DrawableRes iconRes: Int = 0
    ) {
        val snackbar = Snackbar.make(root, "", duration)

        val layout = snackbar.view as SnackbarLayout
        layout.setBackgroundColor(0x00000000) // transparente

        val customView = LayoutInflater.from(root.context)
            .inflate(R.layout.layout_snackbar_zibe, layout, false)

        // Texto
        customView.findViewById<TextView>(R.id.snack_text).text = message

        // Ícono opcional
        val iv = customView.findViewById<ImageView>(R.id.snack_icon)
        if (iconRes != 0) {
            iv.visibility = View.VISIBLE
            iv.setImageResource(iconRes)
        } else {
            iv.visibility = View.GONE
        }

        // Acción opcional
        val tvAction = customView.findViewById<TextView>(R.id.snack_action)
        if (actionText != null && action != null) {
            tvAction.visibility = View.VISIBLE
            tvAction.text = actionText
            tvAction.setOnClickListener { v ->
                action(v)
                snackbar.dismiss()
            }
        } else {
            tvAction.visibility = View.GONE
        }

        // Color fondo card
        customView.backgroundTintList =
            ContextCompat.getColorStateList(root.context, bgColor)

        layout.removeAllViews()
        layout.addView(customView)
        snackbar.show()
    }

    @JvmStatic
    fun showInfo(root: View, message: String) {
        show(
            root = root,
            message = message,
            duration = Snackbar.LENGTH_SHORT,
            bgColor = R.color.colorC,
            actionText = null,
            action = null,
            iconRes = R.drawable.ic_info_24
        )
    }
}
